File: 1431_Kids_With_the_Greatest_Number_of_Candies.java
class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        List kids = new ArrayList();
        int max = candies[0];
        for (int i = 1; i < candies.length; i++) {
            if (candies[i] > max) {
                max = candies[i];
            }
        }
        for (int i = 0; i < candies.length; i++) {
            if (candies[i] + extraCandies >= max) {
                kids.add(true);
            } else {
                kids.add(false);
            }
        }
        return kids;
    }
}

File: 1431_Kids_With_The_Greatest_Number_Of_Candies.py
public class Solution {
    public IList<bool> KidsWithCandies(int[] candies, int extraCandies) {
        bool[] greatest = new bool[candies.Length];
        int max = 0;
        for (int i = 0; i < candies.Length; i++) {
            if (candies[i] > max) max = candies[i];
        }
        for (int i = 0; i < candies.Length; i++) {
            greatest[i] = candies[i] + extraCandies >= max;
        }
        return greatest;
    }
}